TestHostChecker flake (negative WaitGroup counter)

https://travis-ci.org/TykTechnologies/tyk/jobs/235966342

```
=== RUN   TestHostChecker
WARN Legacy path detected! Upgrade to extended.
INFO host-check-mgr: Loading uptime tests...
INFO host-check-mgr: ---> Adding uptime test: http://127.0.0.1:16500/get
INFO host-check-mgr: ---> Adding uptime test: http://127.0.0.1:16501/get
WARN host-check-mgr: [HOST CHECKER MANAGER] Host is DOWN: http://127.0.0.1:16500/get
WARN host-check-mgr: [HOST CHECKER MANAGER] Host is DOWN: http://127.0.0.1:16501/get
panic: sync: negative WaitGroup counter
goroutine 10752 [running]:
sync.(*WaitGroup).Add(0xc4247283f0, 0xffffffffffffffff)
	/home/travis/.gimme/versions/go1.8.1.linux.amd64/src/sync/waitgroup.go:75 +0x1c9
sync.(*WaitGroup).Done(0xc4247283f0)
	/home/travis/.gimme/versions/go1.8.1.linux.amd64/src/sync/waitgroup.go:100 +0x34
github.com/TykTechnologies/tyk.TestHostChecker.func1(0xdf031f, 0x8, 0xd2d560, 0xc424d49480, 0xc424725980, 0x27)
	/home/travis/gopath/src/github.com/TykTechnologies/tyk/host_checker_test.go:89 +0x2a
github.com/TykTechnologies/tyk.(*testEventHandler).HandleEvent(0xc4241165e0, 0xdf031f, 0x8, 0xd2d560, 0xc424d49480, 0xc424725980, 0x27)
	/home/travis/gopath/src/github.com/TykTechnologies/tyk/host_checker_test.go:64 +0x45
created by github.com/TykTechnologies/tyk.fireEvent
	github.com/TykTechnologies/tyk/_test/_obj_test/event_system.go:205 +0x1e1
exit status 2
```

I can't reproduce locally, and that PR should not affect this test at all. Retrying the build fixed it.

This is very likely a race, as `go test -race` isn't happy at all with the host checker code.

--- a/host_checker.go
+++ b/host_checker.go
@@ -16,7 +16,11 @@ const (
 	defaultSampletTriggerLimit = 3
 )
 
-var HostCheckerClient = &http.Client{Timeout: 500 * time.Millisecond}
+var (
+	HostCheckerClient = &http.Client{Timeout: 500 * time.Millisecond}
+
+	hostCheckTicker = make(chan struct{})
+)
 
 type HostData struct {
 	CheckURL string
@@ -84,7 +88,11 @@ func (h *HostUptimeChecker) HostCheckLoop() {
 			}
 		}
 
-		time.Sleep(h.getStaggeredTime())
+		if runningTests {
+			<-hostCheckTicker
+		} else {
+			time.Sleep(h.getStaggeredTime())
+		}
 	}
 	log.Info("[HOST CHECKER] Checker stopped")
 }

--- a/host_checker_test.go
+++ b/host_checker_test.go
@@ -85,6 +85,8 @@ func TestHostChecker(t *testing.T) {
 	spec.Proxy.StructuredTargetList = sl
 
 	var wg sync.WaitGroup
+	// Should receive one HostDown event
+	wg.Add(1)
 	cb := func(em EventMessage) {
 		wg.Done()
 	}
@@ -109,13 +111,7 @@ func TestHostChecker(t *testing.T) {
 		t.Error("Should update host checker check list")
 	}
 
-	// Should receive one HostDown event
-	wg.Add(1)
-	for _, hostData := range GlobalHostChecker.checker.newList {
-		// By default host check should fail > 3 times in row
-		GlobalHostChecker.checker.CheckHost(hostData)
-	}
-
+	hostCheckTicker <- struct{}{}
 	wg.Wait()
 
 	if GlobalHostChecker.IsHostDown(testHttpAny) {
@@ -133,7 +129,7 @@ func TestHostChecker(t *testing.T) {
 		t.Error("Should return only active host", host1, host2)
 	}
 
-	if GlobalHostChecker.checker.checkTimeout != 10 {
+	if GlobalHostChecker.checker.checkTimeout != defaultTimeout {
 		t.Error("Should set defaults", GlobalHostChecker.checker.checkTimeout)
 	}
 

